File: VB/CustomLogonParametersExample.Module/BusinessObjects/Employee.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.ExpressApp.Security.Strategy

Namespace CustomLogonParametersExample.Module.BusinessObjects
	<DefaultClassOptions> _
	Public Class Employee
		Inherits SecuritySystemUser
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private company_Renamed As Company
		<Association("Company-Employees", GetType(Company))> _
		Public Property Company() As Company
			Get
				Return company_Renamed
			End Get
			Set(ByVal value As Company)
				SetPropertyValue("Company", company_Renamed, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/CustomLogonParametersExample.Module/BusinessObjects/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.ExpressApp.Security.Strategy;

namespace CustomLogonParametersExample.Module.BusinessObjects {
    [DefaultClassOptions]
    public class Employee : SecuritySystemUser {
        public Employee(Session session) : base(session) { }
        private Company company;
        [Association("Company-Employees", typeof(Company))]
        public Company Company {
            get { return company; }
            set { SetPropertyValue("Company", ref company, value); }
        }
    }
}
